pkg/controllers/disruption/controller.go

```diff
@@ -194,10 +203,42 @@ func (c *Controller) executeCommand(ctx context.Context, m Method, cmd Command,
 		return multierr.Append(fmt.Errorf("tainting nodes (command-id: %s), %w", commandID, err), state.RequireNoScheduleTaint(ctx, c.kubeClient, false, stateNodes...))
 	}
 
+	select {
+	case <-ctx.Done():
+		return fmt.Errorf("interrupted (command-id: %s)", commandID)
+	case <-c.clock.After(commandValidationDelay):
+	}
+
+	// Revalidate the proposed command after the node has been tainted.
+	// This addresses a race conditions where pods can schedule to the candidate nodes after the command has been finalized, but before the candidates were tainted.
+	candidates, err := GetCandidates(ctx, c.cluster, c.kubeClient, c.recorder, c.clock, c.cloudProvider, m.ShouldDisrupt, c.queue)
+	if err != nil {
+		return fmt.Errorf("validating command, determining candidates, %w", err)
```

Comment:
This leaves nodes tainted, right?

Reply:
Yes, that was a miss. I've swapped out this PR with an alternate implementation that doesn't involve tainting during validation but I'll fix this in my backup branch in case we need to go back.